src/PerformanceManager.js

```diff
@@ -4,17 +4,17 @@ const BigNumber = require('bignumber.js')
 class PerformanceManager extends EventEmitter {
   /**
    * @param priceFeed
-   * @param {BigNumber|undefined} maxPositionSize
-   * @param {BigNumber} allocation
+   * @param maxPositionSize
+   * @param allocation
    */
   constructor (priceFeed, {
     maxPositionSize,
     allocation
   }) {
     super()
 
-    this.maxPositionSize = maxPositionSize
-    this.allocation = allocation
+    this.maxPositionSize = maxPositionSize && new BigNumber(maxPositionSize)
+    this.allocation = allocation && new BigNumber(allocation)
     this.availableFunds = new BigNumber(allocation)
```

Comment:
this.availableFunds = allocation && new BigNumber(allocation || SOME_DEFAULT_VALUE or 0) ? 
Same for `peak` and `trough` as well

Reply:
fixed 69241fc61bbde8154c9ac42d537895391f78701c